source/utilities.cc

```diff
@@ -294,11 +308,15 @@ namespace aspect
 
             if (check_structure)
               {
-                AssertThrow((*n_values_per_key)[field_index] == n_values,
+                AssertThrow(((*n_values_per_key)[field_index] == n_values || n_values == 1),
                             ExcMessage("The key <" + field_names[field_index] + "> in <"+ property_name + "> does not have "
                                        + "the expected number of values. It expects " + std::to_string((*n_values_per_key)[field_index])
                                        + " values, but we found " + std::to_string(n_values) + " values."));
 
+                // Mark the key which has only one value in inputs
+                if (!((*n_values_per_key)[field_index] == n_values))
+                  only_one_value_fields[field_index] = true;
+
```

Comment:
This looks to me like a somewhat complicated solution. Would the following suggestion also work? In this case you would not need to change anything outside of this if-clause, because you simply modify the map as soon as you find an entry with a single value, and therefore all code outside will not even know this was a single-value field to begin with.
```
if (n_values == 1)
{
  const std::string field_name = field_names[field_index];
  const double field_value = parsed_map.find(field_name)->second;
  for (unsigned int i=1; i<(*n_values_per_key)[field_index]; ++i)
    parsed_map.emplace(field_name, field_value);
}

AssertThrow((*n_values_per_key)[field_index] == n_values,
                      ExcMessage("The key <" + field_names[field_index] + "> in <"+ property_name + "> does not have "
                                           + "the expected number of values. It expects " + std::to_string((*n_values_per_key)[field_index])
                                       + " values, but we found " + std::to_string(n_values) + " values."));


```

Reply:
This could work I think. But when I change the "parse_map_to_double_array" I have a supposition that the 'parsed_map' variable is all set.  I took it from the fact that no 'parsed_map.emplace' is called inside this function but it is called many times in the "parse_string_to_map" method. So I just followed. 

As for why not manage this in the "parse_string_to_map" method, I think that is due to the "n_values_per_key" is passed to this "parse_map_to_double_array" but is not passed to "parse_string_to_map" method.

So I think we have three options that we could take:
1. take your advice. But I would hesitate a little since it looks like this function seems to trust that "parse_map" and doesn't want to change that.
2. pass the "n_values_per_key" to the "parse_string_to_map" and change it there
3. write a new function in between.
I think I need you to review this again.

Reply:
Sorry for the delay, I would still be in favor of my original suggestion (option 1. in your list), here is my reasoning:

We created the `parse_string_to_map` function, because the `parse_map_to_double_array` function became too complicated and mixed *reading* the input file with *interpreting* the input file. If we follow this separation than your suggestion of interpreting a single entry as actually multiple values that have the same value falls into the category of *interpreting* the input file. This is why I think it best to keep it inside this function. You are right that there is currently no other line that modifies `parsed_map`, but it is also not a `const` variable so nothing prevents us from doing so. Do you think this is reasonable?

Reply:
Sure, I think this is very clear, thanks for the explanation

Reply:
Do you feel like okay if we ask another question? Why do we have a scope in the first part of this function? I have never seen such a thing before.

Reply:
Just want to let you know that your solution works like a charm. Thanks

Reply:
I believe I have addressed all commends. I have just run every test on docker and it works fine. I think we can review it again. Thanks